src/middleware/auth/user/authDataOnlyMiddleware.ts

```diff
@@ -1,33 +1,28 @@
 // eslint-disable-next-line @typescript-eslint/no-var-requires
-require("dotenv-safe").config();
+const createError = require("http-errors");
 
-import { Request, Response } from "express";
-import { Next } from "compose-middleware";
+import express, { Request, Response, NextFunction } from "express";
 import jwt from "jsonwebtoken";
+import bearerToken from "express-bearer-token";
+import { compose } from "compose-middleware";
 
-const authDataOnlyMiddleware = async (req: Request, res: Response, next: Next) => {
-  // parsing JWT token.
-  const authHeader = req.headers["authorization"];
-  const userAccessToken = authHeader && authHeader?.split(" ")[1];
+const middleware = (req: Request, res: Response, next: NextFunction) => {
+  if (req.token == null) return next(createError(401, "Not authorized"));
 
-  if (userAccessToken == null) return res.sendStatus(401);
+  let admin;
 
-  const TOKEN = process.env.ACCESS_TOKEN_SECRET;
+  try {
+    admin = jwt.verify(req.token, process.env.JWT_SECRET_USER);
+  } catch (err) {
+    console.log(err);
+    next(createError(403, err.message));
```

Comment:
You should either return, or put the rest of the function (line 22 and 23) inside the `try{}`. Otherwise when a catch() both will happen.